vhdbuilder/packer/test/windows-files-check.ps1

```diff
@@ -179,4 +217,162 @@ function Test-ValidateSinglePackageSignature {
+    }
+
+    Process {
+        do {
+            $cnt++
+            try {
+                $ScriptBlock.Invoke()
+                if ($LASTEXITCODE) {
+                    throw "Retry $cnt : $ErrorMessage"
+                }
+                return
+            } catch {
+                Write-Error $_.Exception.InnerException.Message -ErrorAction Continue
+                if ($_.Exception.InnerException.Message.Contains("There is not enough space on the disk. (0x70)")) {
+                    Write-Error "Exit retry since there is not enough space on the disk"
+                    break
+                }
+                Start-Sleep $Delay
+            }
+        } while ($cnt -lt $Maximum)
+
+        # Throw an error after $Maximum unsuccessful invocations. Doesn't need
+        # a condition, since the function returns upon successful invocation.
+        throw 'All retries failed. $ErrorMessage'
+    }
+}
+
+function Install-Containerd {
+    $containerdFileName = [IO.Path]::GetFileName($global:defaultContainerdPackageUrl)
+    $dest = [IO.Path]::Combine("c:\akse-cache\containerd\", $containerdFileName)
+
+    $installDir="c:\program files\containerd"
+    if (!(Test-Path $installDir)) {
+        New-Item -ItemType Directory $installDir -Force | Out-Null
+    }
+    if ($containerdFilename.endswith(".zip")) {
+        Expand-Archive -path $dest -DestinationPath $installDir -Force
+    } else {
+        tar -xzf $dest -C $installDir
+        mv -Force $installDir\bin\* $installDir
+        Remove-Item -Path $installDir\bin -Force -Recurse
+    }
+
+    $newPaths = [Environment]::GetEnvironmentVariable("Path", [EnvironmentVariableTarget]::Machine) + ";$installDir"
+    [Environment]::SetEnvironmentVariable("Path", $newPaths, [EnvironmentVariableTarget]::Machine)
+    $env:Path += ";$installDir"
+
+    $containerdConfigPath = [Io.Path]::Combine($installDir, "config.toml")
+    # enabling discard_unpacked_layers allows GC to remove layers from the content store after
+    # successfully unpacking these layers to the snapshotter to reduce the disk space caching Windows containerd images
+    (containerd config default)  | %{$_ -replace "discard_unpacked_layers = false", "discard_unpacked_layers = true"}  | Out-File  -FilePath $containerdConfigPath -Encoding ascii
+
+    Get-Content $containerdConfigPath
+
+    Start-Job -Name containerd -ScriptBlock { containerd.exe }
+}
+
+function Test-PullImages {
+    Write-Output "Install Containerd."
+
+    Install-Containerd
+
+    Write-Output "Test-PullImages."
+   
+    Write-Output "Pulling images for windows server $windowsSKU" # The variable $windowsSKU will be "2019-containerd", "2022-containerd", ...
+    foreach ($image in $imagesToPull) {
+        Write-Output "Pulling image $image"
+        Retry-Command -ScriptBlock {
+            & crictl.exe pull $image
+        } -ErrorMessage "Failed to pull image $image"
+
+        crictl.exe rmi $image
+    }
+    Stop-Job  -Name containerd
```

Comment:
Can we remove these cleanup steps in the agent runner vms?

Reply:
Removed.